docs/source/examples/lora_finetune.rst

```diff
@@ -300,7 +302,80 @@ A comparison of the (smoothed) loss curves between this run and our baseline ove
     The above figure was generated with W&B. You can use torchtune's :class:`~torchtune.utils.metric_logging.WandBLogger`
     to generate similar loss curves, but you will need to install W&B and setup an account separately.
 
-As an exercise, you can also try running some evaluation tasks or manually inspecting generations
-output by your saved checkpoints (which can be found in :code:`output_dir`).
-You may want to train the model for longer first, as here we only looked at 500 steps
-(which corresponds to about 2% of one epoch of the Alpaca dataset).
+
+Trading off memory and model performance with LoRA
+--------------------------------------------------
+
+In the preceding example, we ran LoRA on two devices. But with LoRA's memory efficiency it is possible
+to run on a single device as well. This can be done via the command:
+
+.. code-block:: bash
+
+    tune run lora_finetune_single_device --config llama2/7B_lora_single_device
+
+On a single device, we may need to be more cognizant of our peak memory. Let's run a couple experiments
+to see our peak memory during a finetune on the Alpaca dataset. We will experiment along two axes:
+first, which model layers have LoRA applied, and second, the rank of each LoRA layer. (We will scale
+alpha in parallel to LoRA rank, as discussed above.)
+
+To compare the results of our experiments, we can evaluate our models on :code:`truthfulqa_mc2`, a task from
+the `TruthfulQA <https://arxiv.org/abs/2109.07958>`_ benchmark for language models. For more details on how to run this and other evaluation tasks
+with torchtune's EleutherAI evaluation harness integration, see our :ref:`End-to-End Workflow Tutorial <eval_harness_label>`.
+
+Previously, we only enabled LoRA for the linear layers in each self-attention, but in fact there are other linear
+layers we can apply LoRA to: MLP layers and our model's final output projection. Note that for Llama-2-7B the final output
+projection maps to dimension 32000, so enabling LoRA for this layer will increase our peak memory a bit more
+than the other layers. We can make the following changes to our config:
+
+.. code-block:: yaml
+
+  # Model Arguments
+  model:
+    _component_: lora_llama2_7b
+    lora_attn_modules: ['q_proj', 'k_proj', 'v_proj', 'output_proj']
+    apply_lora_to_mlp: True
+    apply_lora_to_output: True
+  ...
+
+.. note::
+    All the below finetunes use the `llama2/7B_lora_single_device <https://github.com/pytorch/torchtune/blob/ada52240514bb9fa07f91a50ad0a31063f13834c/recipes/configs/llama2/7B_lora_single_device.yaml>`_
```

Comment:
```suggestion
    All the finetuning runs below use the `llama2/7B_lora_single_device <https://github.com/pytorch/torchtune/blob/ada52240514bb9fa07f91a50ad0a31063f13834c/recipes/configs/llama2/7B_lora_single_device.yaml>`_
```

Reply:
nit: can we just use the hyperlink with main instead of hardcoding to a particular commit. So in this case it would be: https://github.com/pytorch/torchtune/blob/main/recipes/configs/llama2/7B_lora_single_device.yaml

Reply:
I thought using a specific commit was safer tbh. I think there are pros and cons to each though, no strong preference as long as we are consistent

Reply:
Since we usually say something like "look here for details", my concern is that the details will be outdated. The downside of linking to main is that the file may not exist at the same location and so the link is broken, but maybe thats better than stale info?

Reply:
Yeah fair point. I'm good to update it to main